src/Symfony/Component/Encryption/EncryptionInterface.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Encryption;
+
+use Symfony\Component\Encryption\Exception\DecryptionException;
+use Symfony\Component\Encryption\Exception\EncryptionException;
+use Symfony\Component\Encryption\Exception\InvalidKeyException;
+
+/**
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ *
+ * @experimental in 6.0
+ */
+interface EncryptionInterface
+{
+    /**
+     * Generates new a key to be used with encryption.
+     *
+     * Don't lose your private key and make sure to keep it a secret.
+     *
+     * @param string|null $secret A secret to be used in symmetric encryption. A
+     *                            new secret is generated if none is provided.
+     *
+     * @throws EncryptionException
+     */
+    public function generateKey(string $secret = null): KeyInterface;
+
+    /**
+     * Gets an encrypted version of the message.
+     *
+     * Symmetric encryption uses the same key to encrypt and decrypt a message.
+     * The key should be kept safe and should not be exposed to the public. Symmetric
+     * encryption should be used when you are sending the encrypted message to
+     * yourself.
+     *
+     * Example: You store a value on disk or in a cookie and don't want anyone else
+     * to read it.
+     *
+     * Symmetric encryption is in theory weaker than asymmetric encryption.
+     *
+     * <code>
+     *     $key = $encryption->generateKey();
+     *     $ciphertext = $encryption->encrypt('input', $key);
+     *     $message = $encryption->decrypt($ciphertext, $key);
+     * </code>
+     *
+     * @param string       $message Plain text version of the message
+     * @param KeyInterface $key     A key that holds a string secret
+     *
+     * @return string Output formatted by Ciphertext
```

Comment:
Given `Ciphertext` is an internal implementation detail (stop me if I'm wrong), I think the contract should not mention it. I'd replace this something briefly explaining what kind of string should be returned

Reply:
I agree, it make sense. I'll update

Reply:
You are correct that `Ciphertext` class should not really be mentioned. But the format of the string falls under or BC promise. So we need to call that string something... ie "Symfony Encryption Token". 

We also need to define (in document, not just code) what a SET is. Should I rename the class `Ciphertext` to `SymfonyEncryptionToken`? After that we can refer to "Symfony Encryption Token" (the format, not the class).

Reply:
Do we really need to specify it more than just `string` after all? Looking at other crypto libs like paragonie/halite or defuse/encryption, no one seems to enforce the format of the encrypted message

Reply:
If we want the final string to be transferable to other non-symfony applications, then yes. 

The format of our string will fall under or BC promise.